optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/AbstractScoreDirector.java

```diff
@@ -379,7 +379,9 @@ public static String explainScore(Score workingScore,
 
     @Override
     public String explainScore() {
-        return explainScore(calculateScore(), getConstraintMatchTotalMap().values(), getIndictmentMap().values());
+        Collection constraintMatchTotalCollection = getConstraintMatchTotalMap().values();
+        Collection indictmentCollection = getIndictmentMap().values();
+        return explainScore(calculateScore(), constraintMatchTotalCollection, indictmentCollection);
```

Comment:
This is the raw typing trick as discussed. Here we have raw values of `Collection`

Reply:
Correct. And i had to use it on all 2 places in entire `core`. :-)
If `InnerScoreDirector` had `<Score_>`, it would go away.

Reply:
But now that I think about it, there is another way. Stay tuned.

Reply:
Meanwhile, I 've found a way to avoid the raw usages:

```
    @Override
    public <Score_ extends Score<Score_>> String explainScore() {
        Map<String, ConstraintMatchTotal<Score_>> constraintMatchTotalMap = getConstraintMatchTotalMap();
        Map<Object, Indictment<Score_>> indictmentMap = getIndictmentMap();
        return explainScore(calculateScore(), constraintMatchTotalMap.values(), indictmentMap.values());
    }
```

The trick is assigning the map to a local variable, typing it correctly, before calling values()